crypto/evp/e_aes.c

```diff
@@ -1929,8 +1936,8 @@ static int s390x_aes_gcm_cleanup(EVP_CIPHER_CTX *c)
 }
 
 # define S390X_AES_XTS_CTX		EVP_AES_XTS_CTX
-# define S390X_aes_128_xts_CAPABLE	1	/* checked by callee */
-# define S390X_aes_256_xts_CAPABLE	1
+# define S390X_aes_128_xts_CAPABLE	0	/* checked by callee */
```

Comment:
??? You want EVP_aes_128_xts() to return the generic one now?

Reply:
The xts cabability is checked down in the (perl)asm ("checked by callee"). Those macros were only present such that the BLOCK_CIPHER_custom/generic macros can expand, and it did not matter if EVP_aes_##keylen##_##mode returned a pointer to s390x_aes_##keylen##_##mode or aes_##keylen##_##mode (based on the capability check), since those structures were equal. So I chose to set the capability define to 1 to indicate HW support.

Now that s390x_aes_##keylen##_##mode and aes_##keylen##_##mode differ (only the former sets the ALWAYS_CALL_INIT flag), the capability define must be set to 0, such that a pointer to the latter is returned for xts.

Same applies to ctr.